cluster/aws/util.sh

```diff
@@ -702,11 +702,49 @@ function kube-down {
   echo "Deleting VPC"
   vpc_id=$($AWS_CMD describe-vpcs | get_vpc_id)
   if [[ -n "${vpc_id}" ]]; then
+    echo "Deleting instances in VPC: ${vpc_id}"
+    instance_ids=$($AWS_CMD --output text describe-instances \
+                            --filters Name=vpc-id,Values=${vpc_id} \
+                                      Name=tag:KubernetesCluster,Values=${CLUSTER_ID} \
+                            --query Reservations[].Instances[].InstanceId)
+    if [[ -n ${instance_ids} ]]; then
+      $AWS_CMD terminate-instances --instance-ids $instance_ids > $LOG
+      echo "Waiting for instances deleted"
+      while true; do
+        instance_states=$($AWS_CMD describe-instances --instance-ids $instance_ids | expect_instance_states terminated)
+        if [[ "$instance_states" == "" ]]; then
+          echo "All instances terminated"
+          break
+        else
+          echo "Instances not yet terminated: $instance_states"
+          echo "Sleeping for 3 seconds..."
+          sleep 3
+        fi
+      done
+    fi
+
     elb_ids=$(get_elbs_in_vpc ${vpc_id})
-    for elb_id in ${elb_ids}; do
-      $AWS_ELB_CMD delete-load-balancer --load-balancer-name=${elb_id}
-    done
+    if [[ -n ${elb_ids} ]]; then
+      echo "Deleting ELBs in: ${vpc_id}"
+      for elb_id in ${elb_ids}; do
+        $AWS_ELB_CMD delete-load-balancer --load-balancer-name=${elb_id}
+      done
+
+      echo "Waiting for ELBs to stop"
+      while true; do
+        elb_ids=$(get_elbs_in_vpc ${vpc_id})
+        if [[ -z "$elb_ids"  ]]; then
+          echo "All ELBs deleted"
+          break
+        else
+          echo "ELBs not yet terminated: $elb_ids"
+          echo "Sleeping for 3 seconds..."
+          sleep 3
+        fi
+      done
+    fi
```

Comment:
BTW, in GCE I'm heavily pushing for the master to tear these down as part of of a finalizer on the master. See the discussion in #4630. So I've been pushing back heavily against similar code on the GCE `util.sh`, because I don't want each provider to have to implement it.